test/JDBC/input/BABEL-2805.sql

```diff
@@ -33,3 +33,135 @@ GO
+),
+temp_table_3
+AS
+(
+    SELECT	temp_table_2.[Audit.Id] [Audit.Id],
+            temp_table_2.[Audit.InsertOnly] [Audit.InsertOnly],
+            temp_table_2.[Audit.Timestamp] [Audit.Timestamp],
+            temp_table_2.[Audit.Action] [Audit.Action],
+            [Columns].[Name] [Audit.ColumnName],
+            CASE WHEN temp_table_2.[Audit.Action] = 'I' THEN NULL
+            WHEN temp_table_2.[Audit.Action] = 'D' AND [Columns].[Name] = 'Id' THEN CAST(temp_table_2.[CurrentRow.Id] AS [VARCHAR](MAX))
+            WHEN temp_table_2.[Audit.Action] = 'D' AND [Columns].[Name] = 'CampaignId' THEN CAST(temp_table_2.[CurrentRow.CampaignId] AS [VARCHAR](MAX))
+            WHEN temp_table_2.[Audit.Action] = 'D' AND [Columns].[Name] = 'Type' THEN CAST(temp_table_2.[CurrentRow.Type] AS [VARCHAR](MAX))
+            WHEN temp_table_2.[Audit.Action] = 'D' AND [Columns].[Name] = 'RangeStart' THEN CAST(temp_table_2.[CurrentRow.RangeStart] AS [VARCHAR](MAX))
+            WHEN temp_table_2.[Audit.Action] = 'D' AND [Columns].[Name] = 'RangeEnd' THEN CAST(temp_table_2.[CurrentRow.RangeEnd] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'Id' THEN CAST(temp_table_2.[PreviousRow.Id] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'CampaignId' THEN CAST(temp_table_2.[PreviousRow.CampaignId] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'Type' THEN CAST(temp_table_2.[PreviousRow.Type] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'RangeStart' THEN CAST(temp_table_2.[PreviousRow.RangeStart] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'RangeEnd' THEN CAST(temp_table_2.[PreviousRow.RangeEnd] AS [VARCHAR](MAX)) END [Audit.OldValue],
+            CASE WHEN temp_table_2.[Audit.Action] = 'D' THEN NULL
+            WHEN [Columns].[Name] = 'Id' THEN CAST(temp_table_2.[CurrentRow.Id] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'CampaignId' THEN CAST(temp_table_2.[CurrentRow.CampaignId] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'Type' THEN CAST(temp_table_2.[CurrentRow.Type] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'RangeStart' THEN CAST(temp_table_2.[CurrentRow.RangeStart] AS [VARCHAR](MAX))
+            WHEN [Columns].[Name] = 'RangeEnd' THEN CAST(temp_table_2.[CurrentRow.RangeEnd] AS [VARCHAR](MAX)) END [Audit.NewValue]
+    FROM temp_table_2
+        CROSS JOIN (VALUES	
+                        (CAST('Id' AS [SYSNAME])),
+                        (CAST('CampaignId' AS [SYSNAME])),
+                        (CAST('Type' AS [SYSNAME])),
+                        (CAST('RangeStart' AS [SYSNAME])),
+                        (CAST('RangeEnd' AS [SYSNAME]))
+                    ) [Columns]([Name])
+)
+SELECT temp_table_3.[Audit.ColumnName],
+temp_table_3.[Audit.OldValue],
+temp_table_3.[Audit.NewValue]
+FROM temp_table_3
+WHERE temp_table_3.[Audit.Id] = @p1
+AND temp_table_3.[Audit.InsertOnly] = @p2;
+GO
+
+-- Regression tests for CROSS JOIN
```

Comment:
Please add the below tests to the upgrade test files. Specifically, we need tests using views and procedures.

Reply:
Added